contracts/core/OptimisticLightClient.sol

```diff
@@ -48,18 +48,21 @@ contract OptimisticLightClient is ILightClient {
     /**
      * @inheritdoc ILightClient
      */
-    function addOpConsensusState(
-        L1Header calldata l1header,
-        OpL2StateProof calldata proof,
-        uint256 height,
-        uint256 appHash
-    ) external override returns (uint256 fraudProofEndTime, bool ended) {
+    function updateClient(L1Header calldata l1header, bytes calldata proof, uint256 height, uint256 appHash)
+        external
+        override
+        returns (uint256 fraudProofEndTime, bool ended)
+    {
         uint256 hash = consensusStates[height];
         if (hash == 0) {
             // if this is a new apphash we need to verify the provided proof. This method will revert in case
             // of invalid proof.
             verifier.verifyStateUpdate(
-                l1header, proof, bytes32(appHash), l1BlockProvider.hash(), l1BlockProvider.number()
+                l1header,
+                abi.decode(proof, (OpL2StateProof)),
```

Comment:
Where does `abi.decode` come from? `abi` is a builtin in solidity that uses the calling contract's ABI to do the decoding/encoding?

Reply:
yep it's a builtin!

Reply:
[here are the sol docs on this](https://docs.soliditylang.org/en/develop/cheatsheet.html#abi-encoding-and-decoding-functions) in case you wanted more details (admittedly they're not the easiest to find)

Reply:
so the `abi.decode` here is just taking the generic proof byte array and returning an `OpL2StateProof` struct

we can add a test case for this, but it should throw an error if the byte array isn't a valid `OpL2StateProof` struct